pkg/controller/service/handlers.go

```diff
@@ -262,24 +319,17 @@ func (s *Service) changePassword(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if _, err := s.users.UpdatePasswordHash(r.Context(),
-		passwordRecoveryToken.UserID, hash.Hash(changePasswordRequest.Password)); err != nil {
+	if _, err := s.internalUsers.UpdateInternalUserPasswordHash(r.Context(),
+		passwordRecoveryToken.InternalUserID, hash.Hash(changePasswordRequest.Password)); err != nil {
 		log.WithError(err).Error("update password hash")
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	user, err := s.users.GetUser(r.Context(), passwordRecoveryToken.UserID)
```

Comment:
What's the motivation for this? Seems nice to have

Reply:
At first I thought we'd be returning the internal user, which I'd like to not have the frontend accidentally form dependencies on if we don't need to. I can just fetch the external user and return it, though.

Reply:
Oh wait, it's because we keep track of whether an internal user has validated their email by adding them to the `users` table. So if a user wants to reset the password to their unverified email, we'll error when trying to fetch the `user` record.